etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java

```diff
@@ -133,7 +133,9 @@ public Map<HttpEndpoint, Function<DomainRequest, DomainResponse>> domainRegistra
                 PartnerMetadataConverter.class, HapiPartnerMetadataConverter.getInstance());
         // Validation rules
         ApplicationContext.register(RuleLoader.class, RuleLoader.getInstance());
-        ApplicationContext.register(RuleEngine.class, RuleEngine.getInstance());
+        ApplicationContext.register(
+                ValidationRuleEngine.class,
```

Comment:
The first value in register should be an interface and the second should be the class that is defining the behavior for an interface

Reply:
As commented above, we need to be able to identify two different implementations of the rule engine

Reply:
Thoughts on doing a parent interface (engine), and two child interfaces that will extend the parent interface (RuleEngine, ValidationEngine)? With this setup, you can use the child interface to register the implementation for each. The only change here is the naming for the parent interface, the child interfaces, and the implementations for each child interface. You register both engines with a unique interface for each.

Reply:
I don't think the value we register in the ApplicationContext needs to be an interface, a lot of the existing ones are not. Is there a benefit in creating the interface even if it will have only one implementation?

Reply:
I think we generally wrap 3rd party dependencies in an interface to reduce dependencies